extensions/builder.md

```diff
@@ -0,0 +1,145 @@
+# Builder
+
+A **builder** is an image that MUST contain an implementation of the lifecycle, and build-time environment, and MAY contain buildpacks. Platforms SHOULD use builders to ease the build process.
+
+## Table of Contents
+
+## Required File/Directories
+A builder MUST have the following directories/files:
+- `/cnb/order.toml` &rarr; As defined in the [platform specification][order-toml-spec]
+- `/cnb/stack.toml` &rarr; As defined in the [platform specification][stack-toml-spec]
+- `/cnb/lifecycle/<lifecycle binaries>` &rarr; An implementation of the lifecycle, which contains the required lifecycle binaries for [building images][lifecycle-for-build].
+
+In addition, every buildpack blob contained on a builder MUST be stored at the following file path:
+- `/cnb/buildpacks/...<buildpack ID>/<buildpack version>/`
+
+## Environment Variables/Labels
+A builder's environment is the build-time environment of the stack, and as such, MUST fulfill all of the [build image specifications][build-image-specs].
+
+The following variables MUST be set in the builder environment (through the image config's `Env` field):
+
+| Env Variable    | Description
+|-----------------|--------------------------------------
+| `WorkingDir`  | The working directory where buildpacks should default (eg: `/workspace`)
+| `CNB_APP_DIR`  | Application directory of the build environment (eg: `/app`)
+| `CNB_LAYERS_DIR`  | The directory to create and store `layers` in the build environment (eg: `/layers`)
+| `CNB_PLATFORM_DIR`  | The directory to create and store platform focused files in the build environment (eg: `/platform`)
+
+The following variables MAY be set in the builder environment (through the image config's `Env` field):
+
+| Env Variable    | Description
+|-----------------|--------------------------------------
+| `SERVICE_BINDING_ROOT`  | The directory where services are bound
+
+The following labels MUST be set in the builder environment (through the image config's `Labels` field):
+
+| Env Variable    | Description
+|-----------------|--------------------------------------
+| `io.buildpacks.builder.api`  | The Builder API this builder implements (at this point, **0.1**)
+| `io.buildpacks.builder.metadata`  | A JSON object representing [Builder Metadata](#iobuildpacksbuildermetadata)
+| `io.buildpacks.buildpack.order`  | A JSON object representing the [order of the buildpacks stored on the builder](#iobuildpacksbuildpackorder)
+| `io.buildpacks.buildpack.layers`  | A JSON object representing the [buildpack layers](#iobuildpacksbuildpacklayers)
+| `io.buildpacks.lifecycle.version`  | A string, representing the version of the lifecycle stored in the builder (greater than `0.9.0`)
+| `io.buildpacks.lifecycle.apis`  | A JSON object representing the [lifecycle APIs](#iobuildpackslifecycleapis) the lifecycle stored in the builder supports
+
+### `io.buildpacks.builder.metadata`
```

Comment:
```suggestion
### `io.buildpacks.builder.metadata` (JSON)
```
We should include `(JSON)` here and in all the other headings to match the other specs.

Reply:
Resolved in 557b34f